attestation-service/attestation-service/src/config.rs

```diff
@@ -65,12 +73,9 @@ impl TryFrom<&Path> for Config {
     ///            "duration_min": 5
     ///        }
     ///    }
-    type Error = anyhow::Error;
-    fn try_from(config_path: &Path) -> Result<Self, Self::Error> {
-        let file = File::open(config_path)
-            .map_err(|e| anyhow!("failed to open AS config file {}", e.to_string()))?;
-
-        serde_json::from_reader::<File, Config>(file)
-            .map_err(|e| anyhow!("failed to parse AS config file {}", e.to_string()))
+    type Error = ConfigError;
+    fn try_from(config_path: &Path) -> Result<Self, ConfigError> {
+        let file = File::open(config_path)?;
```

Comment:
The original `map_err(|e| anyhow!("failed to open AS config file {}", e.to_string()))?;`

will add extra context of the config file path and it is now opening AS config file. I am not sure the directly `?` here will result in a `IO(#[from] std::io::Error)` defined in `ConfigError`, which will only show an error information `io error`. This would make users confused.

My suggestion is that we can add some more context due to error type, s.t.
```rust
pub enum ConfigError {
    #[error("Failed to open CoCoAS' configuration file: {0}")]
    IO(#[from] std::io::Error),
    #[error("Illegal format of the content of the configuration file: {0}")]
    SerdeJson(#[from] serde_json::Error),
}
```

This would make the error information clearer. wdyt?

Reply:
I agree. I don't think we need the `{0}` when wrapping errors, though.